### DISCUSSION TITLE
globbing function

### DISCUSSION THREAD

**Discussion by fnaj77** [janet-lang/janet#865] [Q&A] [answered]
Hi to all,
is there a *nix-like pathname pattern expansion function to filter the array returned by os/dir?
I mean something that finds and returns all the pathnames matching a specified pattern according to the rules normally used in a linux shell (i.e. "*.txt" to select only those files with "txt" extension).

os/dir accepts only a directory path and and I can't find anything like this in spork.

Thank you for the amazing job!

**Comment by sogaiu** [comment]
I don't know of a built-in option.

Don't know if it would be helpful for your use case, but IIUC there is a `glob` function in @andrewchambers' [janet-sh](https://github.com/andrewchambers/janet-sh): https://github.com/andrewchambers/janet-sh/blob/599495ba4099f1aa6ddea67d1102f514f15237cd/_sh.c#L46-L52

**Comment by andrewchambers** [comment, marked as answer]
I also wrote a peg based glob, it is probably missing some functionality, but might be a good thing to look at: https://github.com/andrewchambers/janet-glob

**Reply by fnaj77** [reply]
Thank you for your replies @sogaiu and @andrewchambers.
I'll definitely take a look at @andrewchambers' glob implementation.

**Reply by pepe** [reply]
I just found that the janet-glob does not have `project.janet` file, so I cannot add it to pkgs. May I ask you @andrewchambers to add one, as I think it is a great addition.
